toolkits/landcover/impl/dataset.js

```diff
@@ -48,8 +61,9 @@ var Dataset = function(id, defaultVisParams) {
 Dataset.prototype.filterDate = function(start, end) {
   // TODO(gino-m): Implement month and year ranges.
   // TODO(gino-m): Implement single day/month/year.
-  this.collection_ = this.collection_.filterDate(start, end);
-  return this;
+  var dataset = this.clone_();
```

Comment:
This is gross.  Are there other options?

Reply:
Yes, it's icky. A few other options come to mind:

1. Don't make Dataset immutable, and continue treating Datasets as mutable [Builders](https://medium.com/@axelhadfeg/builder-pattern-using-javascript-and-es6-ec1539182e24) (link is for ES6, but same idea applies)
2. Add all state/fields to constructor and call here (worse).
3. Implement a util to clone the class dynamically (messy).

Any preference?